core/workflow/aggregatorrole.go

```diff
@@ -64,6 +65,42 @@ func (r *aggregatorRole) UnmarshalYAML(unmarshal func(interface{}) error) (err e
 	return
 }
 
+func (r *aggregatorRole) MarshalYAML() (interface{}, error) {
+	// Aux struct to cast Roles as an array
+	type rootWorkflow struct {
+		Name     string                   `yaml:"name"`
+		Defaults gera.StringMap           `yaml:"defaults"`
+		Roles    []map[string]interface{} `yaml:"roles"`
+	}
+
+	// TODO: Handle multiple Roles
+	var finalAux []map[string]interface{}
```

Comment:
I suspect you don't need this `finalAux` trick. Your output of `aggregator.MarshalYAML` is an `interface{}` with a `map[string]interface{}` inside. You correctly type-assert this and call it `mapAggregator`. This should be enough, because you'll have a map whose only key is `roles`, with a value of type `interface{}` and true type `[]Role`.

You shouldn't have to cast all items to `[]Role` or `[]map[string]interface{}`, because the kv-copy on L90 should be sufficient, and you should then be able to simply output `aux`.

Reply:
Indeed, that was my first implementation as well. But it poses a couple of problems:
- if I use the type asserted `map[string]interface{}` as the value for `output.Roles` I get a YAML file with: 
```yaml
roles:
  name:
  connect:
  roles:
    - name:
      connect:
```
Trying to use this to create an env in `coconut` causes an error saying: `cannot unmarshal !!map to workflow._roleUnion`. After having a look at `_roleUnion`, it seems to me that the first `roles` field needs to be an array. Which means we need:
```yaml
roles:
  - name:
    connect:
    roles:
      - name:
        connect:
```
Even if the first `roles` field has a single element. If I try to marshal a `map[string]interface{}` it results in the first output. Using this trick with a `[]map[string]interface{}` results in the second. It is also why an auxiliary `_aggregatorRole` struct is needed. 

This was a hack to make the final YAML mimic the desired one more closely. Let me know if there's a better approach.

Reply:
Note that I did not suggest to use the type-asserted `mapAggregator := auxAggregator.(map[string]interface{})` as the value of `output.Roles`, I suggested to use `mapAggregator["roles"]`, of type `interface{}` as the value of `output["roles"]` (where `output` is a `map[string]interface{}` too). Did this approach result in trouble?